# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/PerformanceObserver.webidl
# Commit: e04ef5171459
# Full Hash: e04ef517145991e2aee797e935f43b7be35ff7aa
# Author: Will Hawkins <whawkins@mozilla.com>
# Date: 2019-04-09 09:54:40
# Regressor Bug: 1539006
# File Overlap Count: 1
# Description:
#   Bug 1539006: Properly implement Performance Timeline Level 2 w3c spec. r=baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D25426
# ==============================================================================

diff -r ffaff6fb1fe3 -r e04ef5171459 dom/webidl/PerformanceObserver.webidl
--- a/dom/webidl/PerformanceObserver.webidl	Mon Apr 08 19:00:31 2019 +0000
+++ b/dom/webidl/PerformanceObserver.webidl	Tue Apr 09 00:43:10 2019 +0000
@@ -8,8 +8,9 @@
  */
 
 dictionary PerformanceObserverInit {
-  required sequence<DOMString> entryTypes;
-  boolean buffered = false;
+  sequence<DOMString> entryTypes;
+	DOMString type;
+  boolean buffered;
 };
 
 callback PerformanceObserverCallback = void (PerformanceObserverEntryList entries,
@@ -19,7 +20,8 @@
  Constructor(PerformanceObserverCallback callback),
  Exposed=(Window,Worker)]
 interface PerformanceObserver {
-    void observe(PerformanceObserverInit options);
+    [Throws] void observe(optional PerformanceObserverInit options);
     void disconnect();
     PerformanceEntryList takeRecords();
+    static readonly attribute object supportedEntryTypes;
 };